src/Development/IDE/Core/Rules.hs

```diff
@@ -142,9 +141,12 @@ getHomeHieFile f = do
   mbHieTimestamp <- use GetModificationTime normal_hie_f
   srcTimestamp   <- use_ GetModificationTime f
 
-  let isUpToDate
-        | Just d <- mbHieTimestamp = comparing modificationTime d srcTimestamp == GT
-        | otherwise = False
+  let isUpToDate = case (modificationTime <$> mbHieTimestamp, modificationTime srcTimestamp) of
+          -- If the src file is in VFS, then we know it must have been typechecked
+          -- by now (due to kick) and therefore its .hie file is up to date
```

Comment:
Is this true, What happens if you just opened the file? I don’t see what guarantees that the .hie file has been generated before we get here.

Reply:
Hmm, you are right, we don't have a dependency on the `Typecheck` rule here. Looking at the rule for `GetModIface`, there are a couple of different scenarios depending on whether the module is a file of interest or not, etc. As I'm not keen on duplicating the logic here, I am going to declare a dependency on `GetModIface` as a tight overapproximation of what is needed here.